Suspicious code in LazyFutureStream.lazyFutureStream()

The [method](https://github.com/aol/cyclops-react/blob/15a0f62fae617c74ff188653f54ec446a7ef7b6c/src/main/java/com/aol/cyclops/types/futurestream/LazyFutureStream.java#L3441) has two identical sequential conditionals with different logics. Most likely the second is incorrect.

```java
static <T> LazyFutureStream<T> lazyFutureStream(Stream<T> stream) {
        if (stream instanceof LazyFutureStream)
            return (LazyFutureStream<T>) stream;
        if (stream instanceof LazyFutureStream)
            stream = ((LazyFutureStream) stream).toQueue()
                                                .stream(((LazyFutureStream) stream).getSubscription());
        LazyReact react = new LazyReact(
                                        ThreadPools.getCurrentThreadExecutor(), RetryBuilder.getDefaultInstance()
                                                                                            .withScheduler(ThreadPools.getSequentialRetry()),
                                        false, new MaxActive(
                                                             1, 1));
        return new LazyFutureStreamImpl<T>(
                                           react, stream);

    }
```

@johnmcclean-aol 
I tried `gradle test` and it has failed with lots of:
```
~/cyclops-react/src/main/java/com/aol/cyclops/control/Matchable.java:1573: error: incompatible types: cannot infer type-variable(s) T,V,X
                                         .inCaseOfManyType(predicate, i -> then.get(), predicates)).withType(clazz);
                                                          ^
    (argument mismatch; Function<CAP#1,CAP#2> cannot be converted to Function<? super Object,? extends CAP#3>)
  where T,V,X,R are type-variables:
    T extends Object declared in method <T,V,X>inCaseOfManyType(Predicate,Function<? super T,? extends X>,Predicate<V>...)
    V extends Object declared in method <T,V,X>inCaseOfManyType(Predicate,Function<? super T,? extends X>,Predicate<V>...)
    X extends Object declared in method <T,V,X>inCaseOfManyType(Predicate,Function<? super T,? extends X>,Predicate<V>...)
    R extends Object declared in class CheckValues
  where CAP#1,CAP#2,CAP#3 are fresh type-variables:
    CAP#1 extends Object super: Object from capture of ? super Object
    CAP#2 extends CAP#3 from capture of ? extends CAP#3
    CAP#3 extends R from capture of ? extends R
```
Do I miss anything?

--- a/src/main/java/com/aol/cyclops/types/futurestream/LazyFutureStream.java
+++ b/src/main/java/com/aol/cyclops/types/futurestream/LazyFutureStream.java
@@ -3441,9 +3441,6 @@ public interface LazyFutureStream<U> extends Functor<U>, Filterable<U>, LazySimp
     static <T> LazyFutureStream<T> lazyFutureStream(Stream<T> stream) {
         if (stream instanceof LazyFutureStream)
             return (LazyFutureStream<T>) stream;
-        if (stream instanceof LazyFutureStream)
-            stream = ((LazyFutureStream) stream).toQueue()
-                                                .stream(((LazyFutureStream) stream).getSubscription());
         LazyReact react = new LazyReact(
                                         ThreadPools.getCurrentThreadExecutor(), RetryBuilder.getDefaultInstance()
                                                                                             .withScheduler(ThreadPools.getSequentialRetry()),

